### DISCUSSION TITLE
Backfilling unit tests?

### DISCUSSION THREAD

**Discussion by piofusco** [mozilla-mobile/firefox-ios#9497] [General]
There are a number of classes within the repo that do not have unit tests. For some of these classes, there are higher order tests that interact with the UI and probably cover a lot of their functionality, but there's a lot of behavior left unchecked. While I wouldn't suggest having a unit test just to check a box, I am wondering if it would be appropriate to add unit tests for more critical or complicated classes. One class that comes to mind is `AppDelegate`. 

Would it be appropriate for someone to backfill unit tests for some of the more involved classes? Additionally, is there any intention to increase unit testing coverage over time?

**Comment by lmarceau** [comment]
Hi @piofusco, that's a great suggestion. We're all in favour of having more tests coverage 👍 If you see fit for more unit tests in particular areas, you're more than welcome to do so. At the moment we're more focused on adding coverage on new classes only since we have a lot of work in our pipeline. Keep in mind for the AppDelegate that there's currently a PR opened to move to SceneDelegate thought, so if you wanna touch that are I would suggest having a look [there](https://github.com/mozilla-mobile/firefox-ios/pull/9464/files).

**Reply by piofusco** [reply]
Awesome - thanks for such a quick response! I appreciate the feedback. Sounds like any tests around AppDelegate/SceneDelegate might have to wait. I'll see if I can find a lower hanging fruit and put together an MR. Looking forward to the feedback! If there are any classes you have in mind, I am definitely open to suggestions.

**Reply by lmarceau** [reply]
I don't have any suggestions ready since I'm still kind of new on the project 😄 But I would say, have a look at the current code coverage in Client, see which class has some unit tests but not full coverage, and then try to supplement those existing unit tests. It's easier to get started on unit tests if there's already some basics in place. Or you could also have a look in some base classes like extensions/helpers and such, always a good place to look at too. That's what I did recently on [this PR](https://github.com/mozilla-mobile/firefox-ios/pull/9452), maybe you can get inspired from it haha.

**Reply by piofusco** [reply]
Sweet. Thanks for the suggestion! I'll try and put something together soon.
